test/cql-pytest/test_shedding.py

```diff
@@ -31,12 +32,47 @@ def table1(cql, test_keyspace):
 # 5. Hence, a 30MiB request will be estimated to take around 60MiB RAM,
 #    which is enough to trigger shedding.
 # See also #8193.
-@pytest.mark.skip(reason="highly depends on configuration")
+#
+# Big request causes the entire connection to be closed (see #8800 for the reasons),
+# this results in NoHostAvailable on the client.
+# We check that there are no unexpected protocol_errors using Scylla Prometheus API.
+# Such errors occurred before, when before closing the connection, the remaining
+# bytes of the current request were not read to the end and were treated as
+# the beginning of the next request.
+#
+# Have no idea why, but on CI the actual memory limit for CQL requests is ~100MiB,
+# so we use ~60MiB request (which is estimated to take ~120MiB according to the logic above)
+# to guarantee shedding.
 def test_shed_too_large_request(cql, table1, scylla_only):
-    prepared = cql.prepare(f"INSERT INTO {table1} (p,t1,t2,t3,t4,t5,t6) VALUES (42,?,?,?,?,?,?)")
-    a_5mb_string = 'x'*5*1024*1024
-    with pytest.raises(InvalidRequest, match=re.compile('large', re.IGNORECASE)):
-        cql.execute(prepared, [a_5mb_string]*6)
+    def get_protocol_errors():
+        result = 0
+        for line in requests.get(f'http://{cql.cluster.contact_points[0]}:9180/metrics').text.split('\n'):
+            if not line.startswith('scylla_transport_cql_errors_total') or 'protocol_error' not in line:
+                continue
+            result += int(line.split(' ')[1])
+        return result
+
+    # protocol_errors metric is always non-zero, since the
+    # cassandra python driver use these errors to negotiate the protocol version
+    protocol_errors_before = get_protocol_errors()
+
+    # separate session is needed due to the cql_test_connection fixture,
+    # which checks that the session is not broken at the end of the test
```

Comment:
I missed this in my original review. Basically, your observation here is the driver DOESN'T KNOW how to handle this failure - a broken connection remains behind and if the driver reuses it, it doesn't work. Doesn't this mean that this whole "shedding" feature is a broken feature?
One can argue that this is a driver bug, not a Scylla bug, but it still means that this is an unusable Scylla feature, doesn't it?

Reply:
Good point.
* In production Scylla clusters there are usually more than one node and the driver should connect to them, but they will probably respond in the same way. It's the client's problem and the client should deal with it.
* The point of closing the connection in this case was to reduce damage to other decent clients and not to consume excessive Scylla resources. This has been achieved.

Reply:
I see. The second point makes sense. Perhaps the comment should have explained this differently (it's not a test framework problem that you need to overcome, but a driver bug), and a driver issue should have been opened, but I guess it is still better keeping a half-broken connection and letting other clients use it too.